src/App.jsx

```diff
@@ -2,19 +2,67 @@ import './App.css';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 import { useState, useEffect } from 'react';
+import axios from 'axios';
 
-const useSemiPersistentState = () => {
-  const [value, setValue] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(value));
-  }, [value]);
-  return [value, setValue];
-};
+const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+const bearerToken = `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`;
 
 const App = () => {
-  const [todoList, setTodoList] = useSemiPersistentState();
-  const addTodo = (newTodo) => {
-    setTodoList([newTodo, ...todoList])
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const fetchData = async () => {
+    const options = {
+      headers: { Authorization: bearerToken },
+    };
+
+    try {
+      const response = await axios.get(url, options);
+      if (response.status !== 200) throw new Error(`${response.status}`);
+      const data = response.data.records;
+      const todos = data.map(todo => todo = { title: todo.fields.title, id: todo.id });
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      setIsError(true);
+      setErrorMessage('Oopsie! Error getting the list. Please refresh the page or try again later')
+      setIsLoading(false);
+      console.log(`Error getting data from the database ${error.message}`);
+    }
+  };
+
+  useEffect(() => {
+    new Promise((resolve, reject) => { setTimeout(() => resolve(fetchData()), 2000) });
```

Comment:
Why are we doing a timeout here? Wouldn't we want to fetch the data and give it to the user as fast as possible?

Reply:
Got it! I had a wrong understanding why we had the setTimeout in the first place - to _emulate_ the db loading. Now that there is a real db loading we do not need it. Fixed.